orchestra/static/orchestra/task/js/required_fields.js

```diff
@@ -0,0 +1,90 @@
+(function() {
+  'use strict';
+
+  var serviceModule =  angular.module('orchestra.task.services');
+
+  /**
+   * Service to keep track of all required fields in a task view and
+   * validate them on submit.
+   */
+  serviceModule.factory('requiredFields', function($rootScope, orchestraService) {
+    var requiredFields = {
+      fields: [],
+      invalid: [],
+      setup: function(data) {
+        this.data = data;
+      },
+      require: function(fields) {
+        this.fields = this.fields.concat(fields)
+      },
+      validate: function() {
+        var requiredFields = this;
+        requiredFields.invalid = [];
+        this.fields.forEach(function(field) {
+          var obj = requiredFields.data;
+          var keys = field.split('.');
+          for (var i in keys) {
+            var key = keys[i];
+            obj = obj[key];
+            if (!obj) {
+              requiredFields.invalid.push(field)
+              break;
+            }
+          }
+        })
+        $rootScope.$broadcast('orchestra:task:validatedFields');
```

Comment:
curiously, if there's a broadcast mechanism, did we need to have our own pre/post signals?  they're useful now because they can return false, but i wonder if we needed them otherwise?